tools/test.py

```diff
@@ -140,9 +142,21 @@ def main():
             empty_cache=empty_cache)
 
     if rank == 0 and 'eval_result' in outputs[0]:
+        stats = dataset.evaluate(outputs)
+
+        # evaluate feature based metrics
+        features = stats.pop('_inception_feat', None)
+        for metric in EvalIterHook.feature_based_metric:
+            if metric in stats:
+                # since there is no parameters or network in FID and KID,
+                # we can directly build a new metric
+                metric_implement = build_metric(stats[metric])
+                assert features is not None
+                X, Y = features
+                stats[metric] = metric_implement(X, Y)
+
```

Comment:
Because the `tools/test.py` script file doesn't use `EvalIterHook`.
However, on second thought, the code to build and compute FID or KID in `EvalIterHook.evaluate` can be moved to `BaseSRDataset.evaluate`.

Reply:
We are testing this on this [PR](https://github.com/KKIEEK/mmediting/pull/8).